### PR TITLE
Local render

### PR DESCRIPTION
**Pull request by rihorn2**
image
Render is inline html.
This enables dashboard in remote compute environments.
It enables saving the dashboard in a notebook and saving that cell output.
Also added was a link to open in new tab in local, making use of the existing codepath
 **---Not enabled ---**
What if/ ice in remote compute. This may involve one solution per cloud provider (can prioritize azure work)
Followup work with Azure notebook to display using their html renderer

### CODE REVIEW

File: python/interpret_community/widget/explanation_dashboard.py [hunk 1]
```diff
@@ -116,31 +125,7 @@ def explanation_visual(id):
             # if there is no internet or the static file exists, use that
             # main_js='http://127.0.0.1:5000/static/index.js'
```
Comment (imatiach-msft) [inline python/interpret_community/widget/explanation_dashboard.py:126]: should this comment be updated? 127.0.0.1

File: python/interpret_community/widget/explanation_dashboard.py [hunk 2]
```diff
@@ -168,25 +159,58 @@ def __init__(self, explanation, model=None, *, dataset=None,
                 raise e
         ExplanationDashboard.service.use_cdn = use_cdn
         ExplanationDashboard.model_count += 1
-        predict_url = "http://{0}:{1}/{2}/predict".format(
-            ExplanationDashboard.service.ip,
-            ExplanationDashboard.service.port,
-            str(ExplanationDashboard.model_count))
-        ExplanationDashboard.explanations[str(ExplanationDashboard.model_count)] =\
+        if (not in_cloud_env):
+            predict_url = "http://{0}:{1}/{2}/predict".format(
```
Comment (imatiach-msft) [inline python/interpret_community/widget/explanation_dashboard.py:163]: have you tried the notebook vm trick to make this work there?

File: python/interpret_community/widget/js/main.js [hunk 3]
```diff
@@ -3,36 +3,39 @@ import ReactDOM from 'react-dom';
...
+      probabilityY={data.probabilityY}
+      trueY={data.trueY}
+      precomputedExplanations={{
+        localFeatureImportance: data.localExplanations,
+        globalFeatureImportance: data.globalExplanation,
+        ebmGlobalExplanation: data.ebmData
+      }}
+      requestPredictions={data.predictionUrl !== undefined ? generatePrediction : undefined}
+      locale={data.locale}
+      key={new Date()}
```
Comment (imatiach-msft) [inline python/interpret_community/widget/js/main.js:37]: what is the key used for?

File: python/interpret_community/widget/templates/inlineDashboard.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<!doctype html>
```
Comment (imatiach-msft) [inline python/interpret_community/widget/templates/inlineDashboard.html:1]: what is the new file above:  python/interpret_community/widget/static/v0.2.js
is that needed?

File: python/interpret_community/widget/explanation_dashboard.py (hunk 2 above)
Comment (imatiach-msft) [reply python/interpret_community/widget/explanation_dashboard.py:163]: nvm, read the intro to PR, it looks like that's not part of this PR

File: python/interpret_community/widget/js/main.js [hunk 5]
```diff
@@ -3,36 +3,39 @@ import ReactDOM from 'react-dom';
...
-    }
-    return Promise.reject(new Error(resp.statusText))
-  }).then(json => {
-    if (json.error !== undefined) {
-      throw new Error(json.error)
-    }
-    return Promise.resolve(json.data)
-  })
+  ReactDOM.render(<ExplanationDashboard
+      modelInformation={{modelClass: 'blackbox'}}
```
Comment (imatiach-msft) [inline python/interpret_community/widget/js/main.js:24]: how do we know this is always blackbox?

File: python/interpret_community/widget/templates/inlineDashboard.html (hunk 4 above)
Comment (imatiach-msft) [reply python/interpret_community/widget/templates/inlineDashboard.html:1]: I guess this is used by the CDN somehow?

File: python/interpret_community/widget/js/main.js (hunk 5 above)
Comment (rihorn2) [reply python/interpret_community/widget/js/main.js:24]: its not used at this time, the explanations don't log their type, this is a place to support model type in the future

File: python/interpret_community/widget/templates/inlineDashboard.html (hunk 4 above)
Comment (rihorn2) [reply python/interpret_community/widget/templates/inlineDashboard.html:1]: it was for uploading to the cdn, not meant to be permanent

File: python/interpret_community/widget/js/main.js (hunk 3 above)
Comment (rihorn2) [reply python/interpret_community/widget/js/main.js:37]: its a required property for react component. Just needs to be unique on reload.

File: python/interpret_community/widget/explanation_dashboard.py (hunk 2 above)
Comment (rihorn2) [reply python/interpret_community/widget/explanation_dashboard.py:163]: Yeah, I want that to work, but it looks like it might not be trivial....

**Pull request merged**